rmgpy/tools/uncertainty.py

```diff
@@ -341,37 +326,65 @@ def loadModel(self, chemkinPath, dictionaryPath, transportPath=None):
                                                               dictionaryPath=dictionaryPath,
                                                               transportPath=transportPath)
 
+    def retrieveSaturatedSpeciesFromList(self,species):
+        """
+        Given a radical `species`, this function retrieves the saturated species objects from a list of species objects.
+        """
+        
+        molecule = species.molecule[0]
+        assert molecule.isRadical(), "Method only valid for radicals."
+        saturatedStruct = molecule.copy(deep=True)
+        saturatedStruct.saturate()
+        for otherSpecies in self.speciesList:
+            if otherSpecies.isIsomorphic(saturatedStruct):
+                return otherSpecies, False
+        else:
```

Comment:
Codacy is flagging it because you don't need the else. It either returns or continues after the for loop. The same thing would be accomplished by removing the else and reducing the indentation of everything after.

Reply:
That's fair. I do agree with @mjohnson541 though, I like having the else there so that it is clear that the code only runs in the event that the for loop does not succeed. Would you object to changing the else statement with a comment like `# else if no species were found, check for the species in the thermo libraries`

Reply:
I think the cleanest way might be to return at the very end of the method. Returning early is also fine, but I would prefer not having the else. A comment would be helpful though.
```
for species in self.speciesList:
    if ... :
        result = something
        break
else:
    result = something_else

return result
```

Reply:
I've removed the else